src/trusted/service_runtime/sel_ldr_standard.c

```diff
@@ -812,72 +812,74 @@ NaClCreateThread(struct NaClAppThread     *natp_parent,
-   * in a register and that's set in NaClStartThreadInApp.
-   */
-  if (NACL_STACK_GETS_ARG) {
-    uint32_t *argloc = p++;
-    *argloc = (uint32_t) NaClSysToUser(nap_child, (uintptr_t) p);
-  }
+    VCHECK(!(stack_ptr & NACL_STACK_ALIGN_MASK),
+           ("stack_ptr not aligned: %016"NACL_PRIxPTR"\n", stack_ptr));
 
-  *p++ = 0;  /* Cleanup function pointer, always NULL.  */
-  *p++ = envc;
-  *p++ = argc;
+    p = (uint32_t *)stack_ptr;
+    strp = (char *)stack_ptr + ptr_tbl_size;
 
-  for (i = 0; i < argc && argv && argv[i]; i++) {
-    *p++ = (uint32_t) NaClSysToUser(nap_child, (uintptr_t)strp);
-    NaClLog(1, "copying arg %d  %p -> %p\n",
-            i, (void *)argv[i], (void *)strp);
-    snprintf(strp, ARG_LIMIT, "%s", argv[i]);
-    strp += argv_len[i];
-  }
-  *p++ = 0;  /* argv[argc] is NULL.  */
+    /*
+     * For x86-32, we push an initial argument that is the address of
+     * the main argument block.  For other machines, this is passed
+     * in a register and that's set in NaClStartThreadInApp.
+     */
+    if (NACL_STACK_GETS_ARG) {
+      uint32_t *argloc = p++;
+      *argloc = (uint32_t) NaClSysToUser(nap_child, (uintptr_t) p);
+    }
 
-  for (i = 0; i < envc && envv && envv[i]; i++) {
-    *p++ = (uint32_t) NaClSysToUser(nap_child, (uintptr_t)strp);
-    NaClLog(1, "copying env %d  %p -> %p\n",
-            i, (void *)envv[i], (void *)strp);
-    snprintf(strp, ARG_LIMIT, "%s", envv[i]);
-    strp += envv_len[i];
-  }
-  *p++ = 0;  /* envp[envc] is NULL.  */
+    *p++ = 0;  /* Cleanup function pointer, always NULL.  */
+    *p++ = envc;
+    *p++ = argc;
 
-  /* Push an auxv */
-  if (nap_child->user_entry_pt) {
-    *p++ = AT_ENTRY;
-    *p++ = (uint32_t)nap_child->user_entry_pt;
-  }
-  *p++ = AT_NULL;
-  *p++ = 0;
+    for (i = 0; i < argc && argv && argv[i]; i++) {
+      *p++ = (uint32_t) NaClSysToUser(nap_child, (uintptr_t)strp);
+      NaClLog(1, "copying arg %d  %p -> %p\n",
+              i, (void *)argv[i], (void *)strp);
+      snprintf(strp, ARG_LIMIT, "%s", argv[i]);
+      strp += argv_len[i];
+    }
+    *p++ = 0;  /* argv[argc] is NULL.  */
+
+    for (i = 0; i < envc && envv && envv[i]; i++) {
+      *p++ = (uint32_t) NaClSysToUser(nap_child, (uintptr_t)strp);
+      NaClLog(1, "copying env %d  %p -> %p\n",
+              i, (void *)envv[i], (void *)strp);
+      snprintf(strp, ARG_LIMIT, "%s", envv[i]);
+      strp += envv_len[i];
+    }
+    *p++ = 0;  /* envp[envc] is NULL.  */
 
-  /* TODO: fix failing CHECK() -jp */
-  if ((char *)p != (char *)stack_ptr + ptr_tbl_size) {
-    NaClLog(1,
-            "p (%p) != stack_ptr (%p) + ptr_tbl_size (%#lx) [%#lx]\n",
-            (void *)p, (void *)stack_ptr, ptr_tbl_size, stack_ptr + ptr_tbl_size);
+    /* Push an auxv */
+    if (nap_child->user_entry_pt) {
+      *p++ = AT_ENTRY;
+      *p++ = (uint32_t)nap_child->user_entry_pt;
+    }
+    *p++ = AT_NULL;
+    *p++ = 0;
+
+    /* TODO: fix failing CHECK() -jp */
```

Comment:
lol, this seems like a TODO from joey but... is this an issue still?

Reply:
I just indented this, I have no idea what this issue is but I'd guess not considering I've never had any problems with it?